lib/msf/core/module/platform_list.rb

```diff
@@ -25,11 +25,13 @@ def self.win32
   # convenient.
   #
   def self.transform(src)
-    if (src.kind_of?(Array))
-      from_a(src)
+    if src.is_a?(String)
```

Comment:
I don't think this should exist in here, is there a way to move it closer to the deserialization process?

Could we just add `attr_reader :platforms` to the `metadata/obj.rb` file directly maybe?